docs/backend/behaviors.md

```diff
@@ -11,12 +11,320 @@ myst:
+  />
+```
+
+After a restart of Plone, the behavior can be added to the content type in the {guilabel}`Content Types` control panel.
+The add and edit forms contain a new field `Subtitle`.
+
+This field is not displayed in most views.
+To display the data entered in this field, you need to modify the page template by adding the field `context.subtitle`.
+
+### Creating a behavior with an adapter and factory
+
+Given you want to display a price with different content types.
+The price is stored as the net value on the type as a floating point number.
+For display, you need at several places the value added tax (VAT) and the gross value.
+
+You create a schema with the net value for the form and attributes for the calculated values.
+You create an adapter to calculate the VAT and gross values.
+You need a marker interface to distinguish between context and adapter.
+
+Add Python code in the file `price.py`:
+
+```python
+from plone.autoform.interfaces import IFormFieldProvider
+from plone.supermodel import model
+from zope import schema
+from zope.interface import Attribute
+from zope.interface import implementer
+from zope.interface import Interface
+from zope.interface import provider
+
+
+@provider(IFormFieldProvider)
+class IPriceBehavior(model.Schema):
+    """Behavior: a price, VAT and gross."""
+
+    price_net = schema.Float(
+        title="Price (net)",
+        required=True,
+    )
+    price_vat = Attribute("VAT 20% of net price")
+    price_gross = Attribute("Price gross (net + VAT 20%")
+
+
+class IPriceMarker(Interface):
+    """Marker for content that has a price."""
+
+
+@implementer(IPriceBehavior)
+class PriceAdapter:
+    def __init__(self, context):
+        self.context = context
+
+    @property
+    def price_net(self):
+        """Getter, read from context and return back"""
+        return self.context.price_net
+
+    @price_net.setter
+    def price_net(self, value):
+        """Setter, called by the form, set on context"""
+        self.context.price_net = value
+
+    @property
+    def price_vat(self):
+        return self.price_net * 0.2
+
+    @property
+    def price_gross(self):
+        return self.price_net + self.price_vat
+
+```
+
+The registration in the `configure.zcml` looks like so:
+
+```XML
+  <plone:behavior
+      factory=".price.PriceAdapter"
+      for=".price.IPriceMarker"
+      marker=".price.IPriceMarker"
+      name="myproject.price"
+      provides=".price.IPriceBehavior"
+      title="Price with net, VAT and gross"
+  />
+  ```
+
+After a restart of Plone, the behavior can be added to the content type in the {guilabel}`Content Types` control panel.
+The add and edit forms contain a new field `Price (net)`.
+
+This field is not displayed in most views.
+To display the data entered in this field, you need to modify the page template by adding the `price_net` field as `context.price_net`.
+To access the `price_vat` and `price_gross` fields, you need to get the adapter in your view class:
```

Comment:
```suggestion
To access the `price_vat` and `price_gross` fields, you need to get the adapted object in your view class:
```

Reply:
in fact its the adapter you need, the adapted object is the `context` of the adapter.

Reply:
What about:
```suggestion
To access the `price_vat` and `price_gross` fields from a browser view, you need to get the adapter to the context of the view:
```
OT: this complexity is the reason why I once wrote a PLIP to add a dexterity/behavior API to `plone.dexterity`, but I never implemented it. Plan was to have an API like `context.behavior['myproject.price'].vat`.

Reply:
The adapter adapts the to be adapted object and returns the adapted object.

Reply:
> The adapter adapts the to be adapted object and returns the adapted object.

![brain-explosion](https://user-images.githubusercontent.com/157140/208128749-eba960f6-ff97-4eb4-9c8a-39216970fb31.gif)

Reply:
Let's adapt the suggestion for English grammar.

```suggestion
To access the `price_vat` and `price_gross` fields from a browser view, you need to get the adapter from the context of the view:
```